### PR TITLE
Resin door rebalance

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Resin doors open instantly
Slightly more HP.
No longer stop the fire
Close delay is reallt quick
Twice the plasma price and twice the building time to prevent door spam

## Why It's Good For The Game

Nowaydays, resin doors have two uses:
 - cancer wall to prevent marines from flaming to the silo, which is boring to dig out for marines and boring to do as a xenos
 - Noob trap to kill xenos that are slowed down by the door
 
 This makes door actually have a interesting value in defense, and not just a way to kill xeno

## Changelog
:cl:
balance: Resin door opens instantly to xenos, more HP, closes automaticly really fast, doesn't stop flamer fire. Twice the plasma price, twice the building time
/:cl:

### CODE REVIEW

**Comment by Glcbitj** [comment]
Why did you almost double the doors HP? That makes them even more of a fucking cancer wall, not just for silo defending but for it's all other uses too.

**Comment by Chaznoodles** [comment]
Your Pools PR is the cause of the first issue, as xenos are not intended to play defensively due to them primarily being forced on short ambushes or quick engagements, and spamming walls and doors is the only way to stop marines from giving pools the ol' weedwacker. Them being fireproof is the only way to stop marines from wooshing away the only viable defensive planning xenos can do for silos, namely spamming walls and doors.

You are slapping bandaids on symptoms to try and make the forced gameplay loop of rush/lose more palatable, without addressing the underlying cause.

**Comment by BraveMole** [comment]
> Your Pools PR is the cause of the first issue, as xenos are not intended to play defensively due to them primarily being forced on short ambushes or quick engagements, and spamming walls and doors is the only way to stop marines from giving pools the ol' weedwacker. Them being fireproof is the only way to stop marines from wooshing away the only viable defensive planning xenos can do for silos, namely spamming walls and doors.
> 
> You are slapping bandaids on symptoms to try and make the forced gameplay loop of rush/lose more palatable, without addressing the underlying cause.

I've seen your cancer wall yesterday, and the one today too. You're almost the last one to do that kind of shit maze, the other xenos builder have adapted and have much more usefull maze, that doesn't lead to xenos defeat.

"the only viable defensive planning xenos can do for silos, namely spamming walls and doors." You're just a shit maze builder, period

**Comment by BraveMole** [comment]
> > Your Pools PR is the cause of the first issue, as xenos are not intended to play defensively due to them primarily being forced on short ambushes or quick engagements, and spamming walls and doors is the only way to stop marines from giving pools the ol' weedwacker. Them being fireproof is the only way to stop marines from wooshing away the only viable defensive planning xenos can do for silos, namely spamming walls and doors.
> > You are slapping bandaids on symptoms to try and make the forced gameplay loop of rush/lose more palatable, without addressing the underlying cause.
> 
> I've seen your cancer wall yesterday, and the one today too. You're almost the last one to do that kind of shit maze, the other xenos builder have adapted and have much more usefull maze, that doesn't lead to xenos defeat.
> 
> "the only viable defensive planning xenos can do for silos, namely spamming walls and doors." You're just a shit maze builder, period

**Comment by Chaznoodles** [comment]
> I've seen your cancer wall yesterday, and the one today too. You're almost the last one to do that kind of shit maze, the other xenos builder have adapted and have much more usefull maze, that doesn't lead to xenos defeat.
> 
> "the only viable defensive planning xenos can do for silos, namely spamming walls and doors." You're just a shit maze builder, period. I've seen plenty of different defensive planning, that allows huge clutch from xenos. And if you place smartly one or two turret at corners, you are golden.

So, underneath the attempt to deflect onto me being 'shit', everything I said was correct?

**Comment by BraveMole** [comment]
> Why did you almost double the doors HP? That makes them even more of a fucking cancer wall, not just for silo defending but for it's all other uses too.

They are still way weaker than resin wall, especially since walls have 50% bullet armor and the door have none

**Review by SplinterGP — CHANGES_REQUESTED** [review]
only reduce the opening time and change the opening animation accordingly to it, preferably make it not use a sleep too

**Comment by usnpeepoo** [comment]
removing the only thing xenos have at delaying flamers? i dont like that

**Comment by BraveMole** [comment]
> So, underneath the attempt to deflect onto me being 'shit', everything I said was correct?

" spamming walls and doors is the only way to stop marines from giving pools the ol' weedwacker. " Spamming doors and wall like you do is a shit strat, it's not even a way to stop marines. Learn to do some real maze, and these use almost no doors.
So no, everything you said was wrong

**Comment by Glcbitj** [comment]
> > So, underneath the attempt to deflect onto me being 'shit', everything I said was correct?
> 
> " spamming walls and doors is the only way to stop marines from giving pools the ol' weedwacker. " Spamming doors and wall like you do is a shit strat, it's not even a way to stop marines. Learn to do some real maze, and these use almost no doors.
> So no, everything you said was wrong

Ok, who gives a shit if you call your wall+door spam a "maze" or some other thing? "bohoo, you need to make the boring ass spam THIS WAY not THAT WAY" It's still a fucking cancerous way of defending the silo "haha just spam structrures that we can build infinitely and easily replace", even if they didn't block the fucking flamers, marines still need to reach the silo to kill it effectively, because if you can't you're probably losing the battleground and are gonna lose the round anyways.

**Comment by BraveMole** [comment]
> Ok, who gives a shit if you call your wall+door spam a "maze" or some other thing? "bohoo, you need to make the boring ass spam THIS WAY not THAT WAY" It's still a fucking cancerous way of defending the silo "haha just spam structrures that we can build infinitely and easily replace".
> Even if they didn't block the fucking flamers, marines still need to reach the silo to kill it effectively, because if you can't you're probably losing the battleground and are gonna lose the round anyways.

Lel, maze were always the only defending tools of xenos for ever. And no, i don't spam doors in my maze, cause they are litteraly useless. The PR make them usefull in a way that is not boring

**Review by SplinterGP — CHANGES_REQUESTED** [review]
you touch code, you improve it, use timers for the sleep and the spawn below it please. on rest its alright then

**Comment by SplinterGP** [comment]
spawn below, just make it a addtimer for same proc with close delay

image

File: code/game/objects/structures/xeno.dm [hunk 1]
```diff
@@ -316,30 +316,32 @@
...
 	isSwitchingStates = 0
-
-	spawn(close_delay)
-		if(!isSwitchingStates && state == 1)
-			Close()
+	addtimer(CALLBACK(src, .proc/Close), close_delay)
 
 /obj/structure/mineral_door/resin/Close()
-	if(!state || !loc)
+	if(!state || !loc ||isSwitchingStates)
```
Comment (SplinterGP) [inline code/game/objects/structures/xeno.dm:331]:
```suggestion
	if(!state || !loc || isSwitchingStates)
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
change your CL

**Comment by BraveMole** [comment]
@SplinterGP

**Comment by BraveMole** [comment]
I've set the auto close delay to 2 seconds rather than 20, might be awfull to fight against as a marine, we will see

File: code/game/objects/structures/xeno.dm [hunk 2]
```diff
@@ -239,8 +239,8 @@
 	icon = 'icons/Xeno/Effects.dmi'
 	hardness = 1.5
 	layer = RESIN_STRUCTURE_LAYER
-	max_integrity = 80
-	var/close_delay = 60 SECONDS
+	max_integrity = 150
```
Comment (BraveMole) [inline code/game/objects/structures/xeno.dm:242]:
```suggestion
	max_integrity = 00
```

**Comment by SplinterGP** [comment]
if this doenst go well just turn doors in what they actually can have some use for, remove autoclosing so you can just use it as a pre-deployed defense you leave open until you need it.

**Comment by Swagile** [comment]
garbage levels of gameplay due to this PR

nothing but door spam and forced movement into door spam then doors close on you near instantly and your essentially dead forever

if i could down vote 3 times i would

**Comment by Wayland-Smithy** [comment]
Ship it?

**Pull request merged**